cmake/Platform/Common/Install_common.cmake

```
@@ -349,6 +349,133 @@ function(ly_setup_subdirectories)
     endforeach()
 endfunction()
 
+#! ly_setup_3p_target: Export enough data such that the script-only mode knows about the runtime dependencies
+#! for the given target.  This function returns a string which is a setup function that can be called
+#! in script only mode to create a fake target for the 3p library that just includes its copy dependencies.
+function(ly_setup_3p_target OUTVAR_GENERATED_DATA_FILENAME OUTVAR_FILE_NAME_TO_GENERATE ALIAS_TARGET_NAME)
+    ly_de_alias_target(${ALIAS_TARGET_NAME} TARGET_NAME)
+    o3de_get_dependencies_for_target(
+            TARGET "${TARGET_NAME}"
+            COPY_DEPENDENCIES_VAR target_copy_dependencies
+            TARGET_DEPENDENCIES_VAR target_target_dependencies
+        )
+
+    # note that we want to actually create and register the aliased target name
+    # but get the data from the de-aliased target.  So except for the call above
+    # we use the aliased target name.
+    string(REPLACE "::" "__" CLEAN_TARGET_NAME "${ALIAS_TARGET_NAME}")  
+
+    unset(copy_dependencies_target_rel_path)
+    unset(copy_dependencies_source_file_path)
+    foreach(conf IN LISTS CMAKE_CONFIGURATION_TYPES)
+        unset(copy_dependencies_${conf})
+    endforeach()
+
+    foreach(dependency_for_target IN LISTS target_target_dependencies)
+        if (NOT TARGET ${dependency_for_target})
+            # Sometimes, things set actual shared libraries as a target dependency instead of a copy dependency.
+            list(APPEND target_copy_dependencies "${dependency_for_target}\n")
+        endif()
+    endforeach()     
+    
+    # note that the copy dependencies are actually in pairs newline delimited, so
+    # file_to_copy_a\nwhere_to_copy_a;file_to_copy_b\nwhere_to_copy_b;...
+    foreach(copy_dependency IN LISTS target_copy_dependencies)
+        string(REPLACE "\n" ";" copy_dep_pair "${copy_dependency}")
+        list(POP_FRONT copy_dep_pair source_filepath destination_relpath)
+        list(APPEND copy_dependencies_source_file_path "${source_filepath}")
+        # a quirk of list append is that if you give it an empty element, it does nothing.
+        # use dummy element (0) to indicate its an empty path
+        if (NOT destination_relpath)
+            set(destination_relpath "0")
+        endif()
+
+        list(APPEND copy_dependencies_target_rel_path "${destination_relpath}")
```

The `set` function can be used with an expanded string to add empty entry without the special case logic to append a dummy element.

The list just needs to be expanded inside of quotes and then a semicolon can be used to append the destination_relpath.

That would add empty entries.
Such as `PhysX;Qt;;Zlib;;ZStd`
```suggestion
        set(copy_dependencies_target_rel_path "${copy_dependencies_target_rel_path};${destination_relpath}")
```

---

Trying this, Doing just the above actually results in an extra ; for the first element, when the initial variable is empty, that is, the first insertion of "xxx" will set the string to ";xxx" which is two elements of a list.

---

note that trying to get around this, like this
```
         if(copy_dependencies_target_rel_path)
            set(copy_dependencies_target_rel_path "${copy_dependencies_target_rel_path};${destination_relpath}")
        else()
            set(copy_dependencies_target_rel_path "${destination_relpath}")
        endif()
```
also fails in the case of where the first to relpaths are empty, since it will fall into the second case repeatedly when it does in fact need to add them together.

Basically, the logic gets a lot more complicated.

---

I still haven't found a simpler way to do this in a robust way compared to just using a dummy.  thoughts?

---

Ok then it is fine to stick with the placeholder path.